pkg/sources/fake/fake.go

```diff
@@ -15,24 +15,19 @@ const (
 
 var startTime = time.Now()
 
-func getBattery(id uint16) msgs.TelemetryData {
+func getCAN(id uint16) msgs.CAN {
 	t := time.Since(startTime)
 	fid := float64(id)
-	voltage := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
-	temperature := float32(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
+	data := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
 
-	return msgs.NewTelemetryData(&msgs.BatteryModule{
-		ID:          id,
-		Voltage:     voltage,
-		Temperature: temperature,
-	})
+	return msgs.NewCAN(id, data)
 }
 
 // GenFake generates fake data
 func GenFake(bus *pubsub.MessageBus) {
 	for {
-		for battery := uint16(1); battery <= 36; battery++ {
-			bus.Publish("battery", getBattery(battery))
+		for msg := uint16(1); msg <= 36; msg++ {
+			bus.Publish("CAN", getCAN(msg))
```

Comment:
Correct me if I'm wrong, but as I see it:

1. A message is read from the receiver, and marshalled into this `msgs.CAN` struct
2. It is then published on the `CAN` topic to all the subscribers
3. Each subscriber (the database and the websocket) checks the CAN ID, and decides what to do with it

I don't think we need to have a separate type for each CAN message type, but it makes sense (in my mind at least) to do the check before we publish (which is only once per message), rather than after it is published (once for each subscriber).